src/func.c

```diff
@@ -1222,6 +1223,49 @@ void FuncDeclaration::semantic(Scope *sc)
 
 void FuncDeclaration::semantic2(Scope *sc)
 {
+    //printf("FuncDeclaration::semantic2 [%s] fd0 = %s %s\n", this->loc.toChars(), this->toChars(), this->type->toChars());
+    if (overnext && !errors)
+    {
+        OutBuffer buf1;
+        mangleToFuncSignature(&buf1, this);
+        const char *s1 = buf1.peekString();
+
+        OutBuffer buf2;
+        for (Dsymbol *s = overnext; s; )
+        {
+            if (FuncDeclaration *fd = s->isFuncDeclaration())
+            {
+                s = fd->overnext;
+                if (fd->errors)
+                    continue;
+
+                // Don't have to check conflict between declaration and definition.
+                if ((fbody != NULL) != (fd->fbody != NULL))
+                    continue;
+
+                buf2.reset();
+                mangleToFuncSignature(&buf2, fd);
```

Comment:
By computing the signature over and over, this can result in terrible compiler performance.

The  type `deco` system, on the other hand, produces a signature only once and uses a hash for lookup.